### PR TITLE
[Toucan 2 prep] Don't invoke Toucan models as functions

### PR DESCRIPTION
**Pull request by camsaul**
Part of https://github.com/metabase/metabase/issues/24914

[Toucan 2](https://github.com/camsaul/toucan2) is _about to drop*_ so this will be the first of a few PRs in preparation for it.

I've mentioned this a few times before but the biggest difference between Toucan 2 and Toucan 1 is that there's no concept of defining models anymore (i.e., no more `defmodel`) -- you just use `defmethod` to tell it to do certain things when you do DB stuff with certain keywords; this also lets you compose behaviors using for models by deriving them from other keywords. I'll demo this all again soon.

Toucan 2 has an optional Toucan 1 compatibility layer so you don't need to rewrite your _entire_ codebase (I added this after my first attempt at porting Metabase to Toucan 2 last year), but some things just had to change. I went back on forth on having the compatibility version of `defmodel` create something that implements `clojure.lang.IFn` so you could continue invoking models like `(Database 1)` but there was no way to get that working without making the actual non-compatibility part of Toucan 1 a big mess. 

I wrote a custom Kondo linter that will complain if you invoke a Toucan model as a function and then went in and fixed everything. Kondo made this change pretty easy 😻

### CODE REVIEW

File: bin/kondo.sh [hunk 1]
```diff
@@ -0,0 +1,11 @@
+#! /usr/bin/env bash
+
+# Convenience for running clj-kondo against all the appropriate directories.
+
+find modules/drivers shared enterprise/backend \
```
Comment (bshepherdson) [inline bin/kondo.sh:5]: Doesn't this miss `.`?

File: test/metabase/models/database_test.clj [hunk 2]
```diff
@@ -211,7 +211,7 @@
                                        :value   "new-password"})
                 (testing " updating the value works as expected"
                   (db/update! Database id :details (assoc details :password-path  "/path/to/my/password-file"))
-                  (check-db-fn (Database id) {:kind    :password
+                  (check-db-fn (db/select-one Database :id id) {:kind    :password
```
Comment (bshepherdson) [inline test/metabase/models/database_test.clj:214]: nit: busted indentation

**Review by bshepherdson — APPROVED** [review]

File: bin/kondo.sh (hunk 1 above)
Comment (camsaul) [reply bin/kondo.sh:5]: that's intentional so we don't search a million `.git` or `node_modules` directories. `src` and `test` are included explicitly in the last line of the script

File: test/metabase/models/database_test.clj (hunk 2 above)
Comment (camsaul) [reply test/metabase/models/database_test.clj:214]: I'll fix this in a follow-on

**Pull request merged**